core/src/test/java/io/grpc/internal/MessageDeframerTest.java

```diff
@@ -211,6 +219,121 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
     verifyNoMoreInteractions(listener);
   }
 
+  @Test
+  public void sizeEnforcingInputStream_readByteBelowLimit() throws IOException {
+    ByteArrayInputStream in = new ByteArrayInputStream("foo".getBytes(Charsets.UTF_8));
```

Comment:
You're always using the same input stream ... maybe create it in a `@Setup` method? This should get rid of the warning as well.